File: board_dsl/src/main/java/com/dooho/board/repository/SearchRepository.java
package com.dooho.board.repository;

import com.dooho.board.entity.SearchEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Repository
@Transactional
public interface SearchRepository extends JpaRepository<SearchEntity, String> {
    List<SearchEntity> findTop10ByOrderByPopularSearchCountDesc();

    boolean existsByPopularTerm(String search);

}

File: board_spring/src/main/java/com/dooho/board/repository/SearchRepository.kt
package com.dooho.board.repository

import com.dooho.board.entity.SearchEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
@Transactional
interface SearchRepository : JpaRepository<SearchEntity?, String?> {
    fun findTop10ByOrderByPopularSearchCountDesc(): List<SearchEntity?>?
    fun existsByPopularTerm(search: String?): Boolean
}